py/py.mk

```diff
@@ -103,6 +103,13 @@ $(BUILD)/$(BTREE_DIR)/%.o: CFLAGS += -Wno-old-style-definition -Wno-sign-compare
 $(BUILD)/extmod/modbtree.o: CFLAGS += $(BTREE_DEFS)
 endif
 
+# External modules written in C.
+ifneq ($(USER_C_MODULES),)
+CFLAGS_MOD += -DMICROPY_HAS_EXTRA_MODULES
+include $(USER_C_MODULES)/*/micropython.mk
+SRC_QSTR += $(BUILD)/genhdr/modules.h
+endif
```

Comment:
How about calling it `cmodules.h` instead of `modules.h`?

And then to eliminate the need for `MICROPY_HAS_EXTRA_MODULES` you could do this:
```
CFLAGS_MOD += -DMICROPY_CMODULES_INCLUDE_H="genhdr/cmodules.h"
```
Then at the top of objmodule.c use:
```c
#ifdef MICROPY_CMODULES_INCLUDE_H
#include MICROPY_CMODULES_INCLUDE_H
#endif
```

And I wouldn't add this `MICROPY_CMODULES_INCLUDE_H` option to mpconfig.h because it's not really settable by the user, it's handled by this Makefile.

Reply:
Good idea! I've applied it.